src/utils/cruder.js

```diff
@@ -1,28 +1,24 @@
-import request from 'request'
 import { tokenVarName, apiUrl } from '../config'
 import { getVar as getEnvVar } from './env'
-import { typeReducer } from './index'
-import { today } from '../utils'
+import { today, typeReducer } from '../utils'
+import hiddenRequire from './hiddenRequire'
 
-const dateParams = [
-  'start_date', 'end_date',
-  'modified_since', 'modified_before'
-]
+const dateParams = ['start_date', 'end_date', 'modified_since', 'modified_before']
 
 /**
  * Makes an authenticated request to the TSheets API.
  * @param {Object} params Token, endpoint, method, body_params.
  * @return {Promise}
  */
-export const makeRequest = (params) => {
+export const makeRequest = ({ params, request = hiddenRequire('request'), apiKey = getEnvVar(tokenVarName) } = {}) => {
```

Comment:
This is where the majority of the API changes spur from. As `request` and `apiKey` can be passed down. It still defaults to the normal node behavior; but it now gives a developer the option to inject.